packages/next/build/webpack-config.ts

```diff
@@ -263,6 +263,26 @@ export default async function getBaseWebpackConfig(
     resolvedBaseUrl = path.resolve(dir, jsConfig.compilerOptions.baseUrl)
   }
 
+  if (isServer) {
+    const packageJsonPath = path.resolve(dir, './package.json')
+    const packageJsonFile = parseJsonFile(packageJsonPath)
```

Comment:
Checking the version of React in their package.json might not be comprehensive enough. They could have `^16.8.6` but if they did their initial installation recently then they might actually have `16.13.1` installed as their React version (due to the version using the `^` which will bump them to the next minor version released).

I think checking their installed dependencies might give you a bit more coverage:

```js
const packagePath = require.resolve('react/package.json')
const { version } = require(packagePath)
```

Reply:
Noted! Will change it!